src/harness/reference_models/pre_iap_filtering/pre_iap_util.py

```diff
@@ -0,0 +1,142 @@
+  return cbsds_within_polygon
+
+def getPpaFrequencyRange(ppa_record, pal_records):
+  """Determines the frequency range of the PPA from the PAL records.
+
+  The first PAL ID is retrieved from the PPA information and the primary frequency
+  assignment of that PAL ID is returned as the PPA frequency range.
+  Args:
+    ppa_record: A dictionary containing PPA record.
+    pal_records: List of PAL record dictionaries.
+  Returns:
+    Operating frequency range of PPA.
+  """
+  pal_id = ppa_record['ppaInfo']['palId'][0]
+
+  for pal_record in pal_records:
+    if pal_record['palId'] == pal_id:
+      return pal_record['channelAssignment']['primaryAssignment']
+  else:
+    # If the for loop completes without identifying the PAL record then raise exception
+    raise Exception("PAL ID: {} could not be found in the PAL_records list".format(pal_id))
+
+def getFssNeighboringCbsdsWithGrants(cbsd_list, fss_record, distance):
+  """ Get the list of all CBSDs that lie within 150 KMs of FSS and have at least one grant.
+
+  Args:
+    cbsd_list:  List of CbsdData dictionaries as defined in the SAS-SAS specification.
+    fss_record: A FSS record dictionary.
+  Returns:
+    List of CBSDs lying within 150 KMs of FSS and having at least one grant.
+  """
+
+  neighboring_cbsds_with_grants = []
+  for cbsd in cbsd_list:
+    distance_km, _, _ = vincenty.GeodesicDistanceBearing(
+        fss_record['record']['deploymentParam'][0]['installationParam']['latitude'],
+        fss_record['record']['deploymentParam'][0]['installationParam']['longitude'],
+        cbsd['registrationRequest']['installationParam']['latitude'],
+        cbsd['registrationRequest']['installationParam']['longitude'])
+
+    if distance_km <= distance and cbsd['grants']:
+        neighboring_cbsds_with_grants.append(cbsd)
+  return neighboring_cbsds_with_grants
+
+def getFssNeighboringGwbl(gwbl_records, fss_records, distance): 
+  """ Get the list of all fss that lie within 150 KMs of GWBL and operating below 3700000000
+
+  Args:
+    fss_records:  List of FSS records dictionary.
+    gwbl_records: List of  GWBL record dictionary.
+  Returns:
+    List of FSSs lying within 150 KMs of GWBL and operating below 3700000000.
+  """
+
+  list_of_fss_neighboring_gwbl = []
+  for gwbl_record in gwbl_records:
+    for fss_record in fss_records:
+
+      distance_km, _, _ = vincenty.GeodesicDistanceBearing(
+          fss_record['record']['deploymentParam'][0]['installationParam']['latitude'],
+          fss_record['record']['deploymentParam'][0]['installationParam']['longitude'],
+          gwbl_record['record']['deploymentParam'][0]['installationParam']['latitude'],
+          gwbl_record['record']['deploymentParam'][0]['installationParam']['longitude'])  
+      if distance_km <= distance and fss_record['record']['deploymentParam'][0]\
+             ['operationParam']['operationFrequencyRange']['highFrequency'] < 3700000000:
+        list_of_fss_neighboring_gwbl.append(fss_record)
```

Comment:
If you reverse the loops (for fss, for gwbl) you could break out of the inner loop here so that you don't end up with duplicate entries in the list.